psi4/src/psi4/scfgrad/jk_grad.cc

```diff
@@ -1533,37 +1553,77 @@ void DFJKGrad::compute_hessian()
                         }
                     }
                 }
+                // c[A] = (A|mn) D[m][n]
+                C_DGEMV('N', np, nso*(size_t)nso, 1.0, Amnp[0], nso*(size_t)nso, Dtp[0], 1, 0.0, cp, 1);
+
+                // First alpha
+                // (A|mj) = (A|mn) C[n][j]
+                C_DGEMM('N','N',np*(size_t)nso,na,nso,1.0,Amnp[0],nso,Cap[0],na,0.0,Aa_mip[0],na);
+                // (A|ij) = (A|mj) C[m][i]
+                #pragma omp parallel for
+                for (int p = 0; p < np; p++) {
+                    C_DGEMM('T','N',na,na,nso,1.0,Aa_mip[p],na,Cap[0],na,0.0,&Aa_ijp[0][p * (size_t) na * na],na);
+                }
+                // Beta
```

Comment:
This chunk of code needs to be moved outside the loops, as [done here](https://github.com/psi4/psi4/pull/1805/files).  It's correct as-is, but veeeeeerrrrrry slow.

Reply:
Think this is done now. David mentioned that you had made a change which provided some massive speedups. I wasn't 100% sure if I'd done this work before or after that, so I'm glad you settled it!